### PR TITLE
67-update-student-group

### LINKED ISSUE #64: [MT] Update Student Group for a List of Students
PUT route. Call the route "update-bulk" and include it in students.js

This route will update the student_group_id field for every given student_id.

Do not update any other fields for a student record.

### PR DESCRIPTION
**Pull request by mattcalwu**
…t the group ID must be specified in the other table otherwise it will not work)

### CODE REVIEW

**Comment by mattcalwu** [comment]
I added a function in utils that returns true or false if a value is a number, I didn't want to modify isNumeric because I was worried that it might mess up other code

File: routes/utils.js [hunk 1]
```diff
@@ -6,6 +6,13 @@ const isNumeric = (value, errorMessage) => {
   }
 };
 
+const isNum = (value) => {
```
Comment (anikait-rao) [inline routes/utils.js]: why do we need this function when we already have isNumeric? I think you can just loop through the array input and call isNumeric on each of them

File: routes/students.js [hunk 2]
```diff
@@ -176,6 +176,33 @@ router.post('/', async (req, res) => {
...
+router.put('/update-bulk', async (req, res) => {
+  try {
+    const { studentIds, studentGroupId } = req.body;
+    if (!isArray(studentIds)) {
+      throw new Error('studentIds must be an Array');
+    }
+    if (studentIds.every(isNum) === false) {
+      throw new Error('studentIds must contain Numbers');
+    }
+    isNumeric(studentGroupId, 'Student Group Id must be a Number');
```
Comment (anikait-rao) [inline routes/students.js]: they should also be allowed to set studentGroupId to `null`

**Review by anikait-rao — CHANGES_REQUESTED** [review]
see comments

**Review by anikait-rao — APPROVED** [review]
looks good

**Pull request merged**